### PR TITLE
Add code coverage tests

### PR DESCRIPTION
**Pull request by xymostech**
This adds code coverage tests using [nyc](https://github.com/bcoe/nyc),
based on #51. This both makes the tests run code coverage, as well as
adding tests to get us up to 100% line and branch coverage!

Test Plan: `npm run test`

@jlfwong @kentcdodds

### CODE REVIEW

**Comment by xymostech** [comment]
(you can see the code coverage running at the end of the travis build: https://travis-ci.org/Khan/aphrodite/builds/129319450)

**Comment by kentcdodds** [comment]
LGTM, though I'd generally recommend not committing changes to the `dist` directory except in specific commits for building and releasing the library. But not really a big deal.

**Comment by kentcdodds** [comment]
Sorry, my enthusiasm was lost in that last comment. This is great! Good work on this. It's fun to see this project continue to evolve and improve!

File: dist/aphrodite.js [hunk 1]
```diff
@@ -179,6 +179,15 @@ module.exports =
 	};
 
 	exports.mapObj = mapObj;
+	// Flattens an array one level
```
Comment (jlfwong) [inline dist/aphrodite.js]: These changes to `dist/aphrodite.js` are unrelated to this diff, right?

File: tests/inject_test.js [hunk 2]
```diff
@@ -72,16 +72,69 @@ describe('injection', () => {
...
+
+            asap(() => {
+                assert.include(styleTag.styleSheet.cssText, ".x{");
+                assert.include(styleTag.styleSheet.cssText, "color:red");
+                done();
+            });
+        });
+
+        it('uses document.getElementsByTagName without document.head', done => {
+            Object.defineProperty(global.document, "head", {
```
Comment (jlfwong) [inline tests/inject_test.js]: lol JavaScript

**Comment by jlfwong** [comment]
Awww yis. Shipit.

**Comment by jlfwong** [comment]
Did you want to add a coverage badge to the README.md too?

File: tests/inject_test.js (hunk 2 above)
Comment (xymostech) [inline tests/inject_test.js]: Yeah it turns out that `document.head` in jsdom is defined using a getter, so I had to do this to override it. ¯\_(ツ)_/¯

File: dist/aphrodite.js (hunk 1 above)
Comment (xymostech) [inline dist/aphrodite.js]: They are. :P

**Comment by xymostech** [comment]
> I'd generally recommend not committing changes to the dist directory except in specific commits for building and releasing the library. But not really a big deal.

Yeah. At KA we use a submodule so it was more useful there before, but at this point we probably shouldn't be updating it unless we make an npm release. I keep doing this because I set up git hooks to automatically build and commit it, but I can turn those off.

> Did you want to add a coverage badge to the README.md too?

I was trying to figure out how to add a coverage badge and couldn't. Soo, if you can figure it out then sure!

**Pull request merged**